File: code/timer2/python/timer_repeat.py
#!/usr/bin/env python3

import threading
import datetime
import time

def print_message(message):
    print(f"{datetime.datetime.now()}: {message}")

def repeat_hello():
    print_message("Hello, World!");
    timer = threading.Timer(2.0, repeat_hello)
    timer.start()

def main():
    timer = threading.Timer(2.0, repeat_hello)
    timer.start()
    print_message("Repeating timer started...");

if __name__ == "__main__":
    main()

File: code/timer2/recipe-01/timer_repeat.cpp
#include "timer.hpp"
#include "print_message.hpp"
#include <chrono>
#include <thread>

void repeat_hello() {
    print_message("Hello, World!");
}

int main() {
    Timer timer;
    timer.start(repeat_hello, std::chrono::seconds{2}, Timer::repeat);
    print_message("Repeating timer started...");
    while (timer.isActive()) {
        std::this_thread::sleep_for(std::chrono::seconds{1});
    }
}
